### PR TITLE
Add attribute lower_extension to CPUContext

### PR DESCRIPTION
**Pull request by reazulhoque**
This PR makes the global `lower_extensions` a part of CPUContext. For Numba IR extension nodes, the pass that adds the extension node can register the method to lower it with the target context.

### CODE REVIEW

**Comment by reazulhoque** [comment]
I fixed the issue flake8 was complaining about.

File: numba/core/typed_passes.py [hunk 1]
```diff
@@ -289,6 +289,14 @@ def run_pass(self, state):
         """
         Convert data-parallel computations into Parfor nodes
         """
+        # Register lowerer for Parfor Node
+        from numba.parfors.parfor_lowering import _lower_parfor_parallel
+        if hasattr(state.targetctx, "lower_extensions"):
+            state.targetctx.lower_extensions[Parfor] = _lower_parfor_parallel
```
Comment (sklam) [inline numba/core/typed_passes.py:295]: Can we avoid assigning to the target context in the middle the pipeline?

Comment (reazulhoque) [reply numba/core/typed_passes.py:295]: I absolutely wanted to, but could not find a better place. Please suggest. My thought process: the pass that adds the `special_extension_ir_node` should also add how to lower it.

Comment (sklam) [reply numba/core/typed_passes.py:295]: I think it make sense for the ``CPUContext.init`` to also insert the ``Parfor`` lowering.

Comment (reazulhoque) [reply numba/core/typed_passes.py:295]: Should we be inserting the `Parfor` lowering in both places? Or should the initialization be removed from this pass and only be happening in `CPUContext.init`?

Comment (sklam) [reply numba/core/typed_passes.py:295]: Yes, remove from this pass and only in `CPUContext.init`.

Comment (reazulhoque) [reply numba/core/typed_passes.py]: Done.

**Review by sklam — APPROVED** [review]
Thanks for the patch!

**Pull request merged**